cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureTemplateBuilderTest.java

```diff
@@ -424,10 +454,50 @@ public void buildWithInstanceGroupTypeGateway() {
         assertTrue(templateString.contains("\"customData\": \"" + base64EncodedUserData(GATEWAY_CUSTOM_DATA) + '"'));
     }
 
+    @Test
+    public void buildWithGatewayInstanceGroupTypeAndLoadBalancer() {
```

Comment:
I know you're working on the public vs. private stuff now. When you have that sorted, you should add test cases for both.

Reply:
I added a test case with both a public and private LB, and verified the expected values were in the ARM template. (Based on my understanding of "expected").